### ISSUE TITLE
Force explicit abi

### ISSUE THREAD

**Issue by maijun-sec** [rust-lang/rust-clippy#13372]
### What it does

This is from a configuration of rustfmt `force_explicit_abi`. 

When using `extern` to specify an external function, it's advised to explicitly specify `C-ABI`.

Although the default is `C-ABI` if `extern` is not specified, it is a convention for Rust to specify it explicitly.

### Advantage

- It is a convention for Rust to specify it explicitly when using `extern` to specify an external function;
- Clippy can support to check such scenario that rustfmt can automatically format.

### Example

bad case:
```rust
// Noncompliant
extern {
    pub static lorem: c_int;
}
```

good case:
```rust
// Compliant
extern "C" {
    pub static lorem: c_int;
}

extern "Rust" {
    type MyType;
    fn f(&self) -> usize;
}
```

**Comment by maijun-sec** [comment]
already have a rustc lint, https://github.com/rust-lang/rust/pull/76219
